TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,246 @@
+     //   this.robot.imu.initialize(parameters);
+        angles = this.robot.imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES);
+        if (degreesToTurn < 0) {
+            while ((angles.firstAngle > degreesToTurn) && this.opMode.opModeIsActive()) {
+                double generalPower = (degreesToTurn - angles.firstAngle)/degreesToTurn;
+                this.robot.motorOne.setPower(generalPower * wheelOnePower);
+                this.robot.motorTwo.setPower(generalPower * wheelTwoPower);
+                this.robot.motorThree.setPower(-generalPower * wheelThreePower);
+                this.robot.motorFour.setPower(-generalPower * wheelFourPower);
+                angles = this.robot.imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES);
+                telemetry.addData("angles", angles.firstAngle);
+                telemetry.addData("degreesToTurn", degreesToTurn);
+                telemetry.update();
+                Log.d("Turning", "angles 164: "  + angles.firstAngle + ", " + angles.secondAngle + ", " + angles.thirdAngle);
+            }
+        } else {
+            while ((angles.firstAngle < degreesToTurn) && this.opMode.opModeIsActive()) {
+                double generalPower = (degreesToTurn - angles.firstAngle)/degreesToTurn;
+                this.robot.motorOne.setPower(-generalPower * wheelOnePower);
+                this.robot.motorTwo.setPower(-generalPower * wheelTwoPower);
+                this.robot.motorThree.setPower(generalPower * wheelThreePower);
+                this.robot.motorFour.setPower(generalPower * wheelFourPower);
+                angles = this.robot.imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES);
+                telemetry.addData("angles", angles.firstAngle);
+                telemetry.addData("degreesToTurn", degreesToTurn);
+                telemetry.update();
+                Log.d("Turning", "angles 176: " + angles.firstAngle + ", " + angles.secondAngle + ", " + angles.thirdAngle);
+            }
+        }
+        this.stop();
+    }
+
+    public void goLeft(double inches) {
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "Robot is moving left");
+
+        startPosition = this.robot.motorOne.getCurrentPosition();
+        endPosition = startPosition - (inches * sideInchToTick); // How far you need to travel
+        while (this.robot.motorOne.getCurrentPosition() > endPosition && this.opMode.opModeIsActive()) {
+            Log.d("going Left", "End Position: " + endPosition);
+            Log.d("going Left", "Start Pos: " + startPosition);
+            Log.d("going Left", "Current Pos: " + this.robot.motorOne.getCurrentPosition());
+
+            this.robot.motorOne.setPower(-wheelOnePower);
+            this.robot.motorTwo.setPower(wheelTwoPower);
+            this.robot.motorThree.setPower(wheelThreePower); // May need to go other direction
```

Comment:
This comment is so spooky. Do we know if this is the right direction? If not we should figure that out before putting this in master